Lib/typing.py

```diff
@@ -1396,8 +1396,15 @@ def _get_protocol_attrs(cls):
 
 
 def _is_callable_members_only(cls):
+    attr_names = _get_protocol_attrs(cls)
+    annotations = getattr(cls, '__annotations__', {})
```

Comment:
What about using `typing.get_type_hints` to resolve annotations for a class?

It will allow having `ClassVar` annotated as str:
```py
@runtime_checkable
class P(Protocol):
    x: 'ClassVar[int]' = 1
```

```suggestion
    annotations = get_type_hints(cls)
```

Reply:
Good question! I'd considered that too but decided against it for two main reasons:
1. `get_type_hints` is slow
2. `get_type_hints` won't work with forward references/ PEP 563 if `ClassVar` is not defined in the caller's namespace

Consider the following:
```python
# foo.py
from __future__ import annotations
from typing import *

@runtime_checkable
class X(Protocol):
 x: ClassVar[int] = 1
 y: SomeUndeclaredType = None
```
```python
# bar.py
from .foo import X
class Y: ...

# Error! get_type_hints cannot resolve 'ClassVar' and 'SomeUndeclaredType'
issubclass(X, Y)
```

Nonetheless, your suggestion has reminded me of a basic workaround for string annotations. Thanks.